tests/integration/targets/rds_instance_modify/tasks/main.yml

```diff
@@ -193,6 +193,63 @@
       - result.changed
       - result.db_instance_identifier == "{{ modified_instance_id }}"
 
+
+  # Test modifying CA certificate identifier -------------------------------------------
```

Comment:
fyi, did not add `check-mode idempotent` and `idempotent` tests as 
even though functionality wise they were working as expected, [due to lines here](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_instance.py#L1537-L1541), it always returns `changed=true`

the reason behind it is that parameter_to_modify always has regardless of it is specified in task or not, something we can look into in another PR.
```
"parameters_to_modify": {
            "AllowMajorVersionUpgrade": true,
            "ApplyImmediately": true,
            "DBInstanceIdentifier": "ansible-test-d62ddfxx7x3d-updated"
        }
```

Reply:
@alinabuzachis nevermind, I have added integration tests
realized that `idempotent` and `check_mode - idempotent` were retruning `changed=True` due to `allow_major_version_upgrade` being set

caused by line here https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_instance.py#L1155-L1157